### PR TITLE
add DMI apis: interface and definition

### PR DESCRIPTION
**Pull request by RyanZhaoXB**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
for the current way, KubeEdge can only get the device connected via mapper and synchronize the device twin, which can not meet the users' requirements for device management. in this pr, we add the apis of DMI including service interface and struct definition. it also update the CRD of the device instance and device model. 

**Which issue(s) this PR fixes**:

Fixes #https://github.com/kubeedge/kubeedge/issues/3760

**Special notes for your reviewer**:
we will implement these apis later in the next submission. 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by gy95** [comment]
need run `hack/update-codegen.sh` and `hack/generate-crds.sh` to update generated codes and crds

File: pkg/apis/devices/v1alpha2/device_model_types.go [hunk 1]
```diff
@@ -25,6 +25,25 @@ import (
...
+	// Required: The device command url.
+	Url string `json:"url"`
+	// Required: The device command method.
+	Method string `json:"method"`
+	// Required: The device command status code.
+	StatusCode []string `json:"status_code"`
+	// +optional
+	Parameters []string `json:"parameters, omitempty"`
+	// +optional
+	Response []interface{} `json:"response, omitempty"`
```
Comment (TianTianBigWang) [inline pkg/apis/devices/v1alpha2/device_model_types.go:46]: maybe just `interface{}` or `[]byte` ?

Comment (RyanZhaoXB) [reply pkg/apis/devices/v1alpha2/device_model_types.go]: ok. i will make it interface{}

**Comment by RyanZhaoXB** [comment]
> need run `hack/update-codegen.sh` and `hack/generate-crds.sh` to update generated codes and crds

i have make it in the latest commit

**Comment by TianTianBigWang** [comment]
lgtm

**Pull request closed without merging**